cl/lib/elasticsearch_utils.py

```diff
@@ -298,14 +434,14 @@ def convert_str_date_fields_to_date_objects(
     :param search_type: The search type to perform.
     :return: None, the function modifies the search results object in place.
```

Comment:
Yeah, this is related to the frontend search for `Parentethicals` where we group results by `cluster_id` in `group_search_results()`

So for this scenario, we get results as a dict from elastic:

```
{
   "key":6,
   "doc_count":2,
   "grouped_by_opinion_cluster_id":{
      "hits":{
         "total":{
            "value":2,
            "relation":"eq"
         },
         "max_score":1.0,
         "hits":[...]
         
}
```
`date` fields contain the date as an `str` e.g: `dateReargued":"1986-01-30`
However, in order to format this date correctly in the frontend, we need to convert it to `datetime.date`.

In Oral arguments, this is not needed since we're not doing grouping there, results are returned as an `AudioDocument(Document)`, and fields in this object have the same data type as defined in the Document mapping.

So I think this is an issue from `django-elasticsearch-dsl` and `elasticsearch-dsl` that doesn't support group representation so it returns the raw response from elastic. Do you think this can be fixed without doing the cast?